src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,28 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { query, status } = useAppSelector(state => state.filter);
+  const setQuery = (newQuery: string) => dispatch(actions.SetQuery(newQuery));
+  const setStatus = (
+    newStatus: Status,
+  ) => dispatch(actions.setStatus(newStatus));
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select
+            data-cy="statusSelect"
+            value={status}
+            onChange={(event) => setStatus(event.target.value as Status)}
```

Comment:
Move these callbacks in `onChange` to the separate functions and then just use it here.
```suggestion
            onChange={changeStatus}
```